### PR TITLE
Adds Dockerfile

### PR DESCRIPTION
**Pull request by obulpathi**
### - What I did
I added a Dockerfile so that users can create a Docker based environment on Mac, Linux or Windows with ease and test their smart contracts in a repeatable way

### - How I did it
I followed the Viper installation instructions from docs/installing-viper.rst and compiled those instructions into Dockerfile

### - How to verify it
Clone the PR brach and do a "docker build -t viper ."
Run "docker run -it viper /bin/bash" then do "viper examples/crowdfund.v.py" to compile the crowdfund smart contract

### - Description for the changelog
Adds the Dockerfile

### - Cute Animal Picture

> 😸

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM ubuntu:17.04
```
Comment (yograterol) [inline Dockerfile]: With `Python:3.6` as base image you saving the rest .

Comment (JackieXu) [inline Dockerfile]: Preferably, use the `alpine` image, there's a Python 3.6 version available. It produces a much smaller image, allowing new people to get started faster.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+MAINTAINER Obul <obulpathi@merkletree.vc>
+
+# force bash always
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+
+# install dependencies
+RUN apt-get update && apt-get install -y git libssl-dev libffi-dev python3.6 python3.6-dev python3-pip
+RUN pip3 install virtualenv
```
Comment (JackieXu) [inline Dockerfile]: Having a `virtualenv` inside a container slightly defeats the purpose. The point of the container is that it's already its own environment.

**Comment by JackieXu** [comment]
I think the current Docker container doesn't simplify things all that much, it suffers from the same issue as the Solidity Docker with regards to getting files into the container. 

Ideally, the Docker image functions as a standalone Viper compiler, giving the user a single command he can run to compile a Viper contract.

File: Dockerfile (hunk 1 above)
Comment (obulpathi) [reply Dockerfile]: Good one. Will do it.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# create virtual environment
+RUN virtualenv --python=/usr/bin/python3.6 --no-site-packages ~/.virtualenvs/viper
+RUN source ~/.virtualenvs/viper/bin/activate
```
Comment (msoedov) [inline Dockerfile]: this dockerfile could be done in 2 lines, without all this crap with virtual env and git clone

**Comment by obulpathi** [comment]
Thank you @yograterol @JackieXu @msoedov for the feedback. I used python:3.6 as the base Docker image and updated the pull request. Can you please take another look to see if anything else needs to be changed. Thanks once again!

**Comment by DavidKnott** [comment]
@obulpathi When I tried running the docker from your branch everything worked up to `viper examples/crowdfund.v.py`, then I got the error:
```
Traceback (most recent call last):
  File "/usr/local/bin/viper", line 4, in <module>
    __import__('pkg_resources').run_script('viper==0.0.2', 'viper')
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 748, in run_script
    self.require(requires)[0].run_script(script_name, ns)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 1524, in run_script
    exec(script_code, namespace, namespace)
  File "/usr/local/lib/python3.6/site-packages/viper-0.0.2-py3.6.egg/EGG-INFO/scripts/viper", line 8, in <module>
ImportError: cannot import name 'parse_to_lll'
```
`make test` worked just fine, has this happened to you?

**Comment by obulpathi** [comment]
@DavidKnott No, it did not happen. Here is a screenshot showing the successful compilation of crowdfund.v.py contract.
viper

I have pushed the docker image to DockerHub: https://hub.docker.com/r/obulpathi/viper/
Can you please give it a try and let me know if you still face the issue? Thanks!

**Comment by obulpathi** [comment]
I removed the python:3.6 base image and rebuilt the viper docker image. Now I see it happening.

**Comment by obulpathi** [comment]
bin/viper:from viper.parser import parse_to_lll

Looks like a bug in bin/viper? parse_to_lll can be imported from viper.parser.parser, while the viper binary is trying to import it from viper.parser

**Comment by DavidKnott** [comment]
I'll make a quick PR to fix this.

**Comment by DavidKnott** [comment]
@obulpathi #387 fixes this.

**Comment by obulpathi** [comment]
@DavidKnott Awesome! Thanks David!

**Comment by DavidKnott** [comment]
@obulpathi Thanks for being patient!  LGTM.

**Pull request merged**